### PR TITLE
[core] Spell damage gives TP

### PR DESCRIPTION
**Pull request by sarca571ca**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
When an entity takes damage from a spell, they gain a flat 50 tp.

## Steps to test these changes
Find a monster that cast spells. Attack them and wait for them to cast on you and note that you gain 50 tp.

### CODE REVIEW

**Review by Xaver-DaRed — APPROVED** [review]

File: src/map/utils/battleutils.cpp [hunk 1]
```diff
@@ -2519,6 +2519,9 @@ namespace battleutils
             // Do we get TP for damaging spells?
             int16 tp = battleutils::CalculateSpellTP(PAttacker, PSpell);
             PAttacker->addTP(tp);
+
+            // Targets of damaging spells gain 50 tp
+            PDefender->addTP((uint16)50);
```
Comment (claywar) [inline src/map/utils/battleutils.cpp]: This cast is misleading, and most likely unnecessary.  `addTP()`'s parameter is a signed int16, not unsigned

Comment (sarca571ca) [reply src/map/utils/battleutils.cpp]: that  makes sense. i had borrowed code from [battleutils.cpp#L2327](https://github.com/LandSandBoat/server/blob/219a712da21f650eddf07ac3d7b0eaa990d9ef96/src/map/utils/battleutils.cpp#L2327) so i just went with what was already used. Even in the start of that ```if statement``` they used int16.

**Comment by zach2good** [comment]
We just thought of some things?
- Has this amount existed forever? Should it be hooked up to a setting for people that want to crank it or turn it off for era/custom purposes?
- Is the 50 amount affected by Store TP etc.?

WTB: Someone go on retail with SAM or some other high-store-TP job and see if you get more TP from getting nuked

**Comment by zach2good** [comment]
Ok, I went to Castle.O on my i119 NIN (lots of Store TP) and let a Yagudo Theologist cast Thunder on me: I got 60TP - so this is affected by Store TP in some way

**Comment by zach2good** [comment]
For reference, I think I have +22 Store TP on at the moment

**Comment by zach2good** [comment]
In other places, I see this being used for giving TP:
```
                PAttacker->addTP(
                    (int16)(tpMultiplier * (baseTp * (1.0f + 0.01f * (float)((PAttacker->getMod(Mod::STORETP) + getStoreTPbonusFromMerit(PAttacker)))))));
```

**Comment by Xaver-DaRed** [comment]
That would suggest that it follows regular store TP equation
`TP/hit = floor( Base TP/hit × (100 + Store TP Total) ÷ 100 )`

base is 50 in this case. your Store TP would give a 1.2 bonus, so 50 * 1.2 = 60

**Comment by zach2good** [comment]
Maths:
```
float storeTPMultiplier = 1.0f + 0.01f * static_cast<float>(PDefender->getMod(Mod::STORETP) + getStoreTPbonusFromMerit(PDefender));
            PDefender->addTP(static_cast<int16>(50 * storeTPMultiplier);
```
```
multiplier = 1.0 + 0.01 * (22 + 0) = 1.22
50 * 1.22 = 61
```
^ I don't actually remember if I had 20 or 22 store TP, but the maths here is correct

**Review by claywar — APPROVED** [review]

**Review by Xaver-DaRed — APPROVED** [review]

**Pull request merged**